### PR TITLE
fix: better type constraints on withPojoToDocument

### PR DESCRIPTION
**Pull request by VolodymyrWork23**

### CODE REVIEW

File: src/mongoose.ts [hunk 1]
```diff
@@ -195,12 +195,12 @@ export function htMongooseFactory(mongoose: any) {
     });
   }
```
Comment (captaincaius) [inline src/mongoose.ts:197]: I don't think TMongooseModel is right - it looks like the constructor is constructing the class itself and not an object thereof.
- [ ] type

Comment (VolodymyrWork23) [reply src/mongoose.ts:197]: done

Comment (captaincaius) [reply src/mongoose.ts:197]: still doesn't look right...
fake it with a class and you'll see.  I think you want an additional generic type parameter that represents the object type that TMongooseModel constructs.
BTW I think I already have a helper interface called Constructor<> somewhere for this (unless I deleted it) - also newer typescript releases may already have it too.

Comment (VolodymyrWork23) [reply src/mongoose.ts:198]: now i using Constructor<>

Comment (captaincaius) [reply src/mongoose.ts:198]: like I said...  TMongooseModel isn't a constructor of TMongooseModel.  The object vs the class.

Comment (VolodymyrWork23) [reply src/mongoose.ts:198]: done

**Pull request merged**